### DISCUSSION TITLE
Standardizing Regression Slopes

### DISCUSSION THREAD

**Discussion by BarthMaximilian** [jazznbass/scan#153] [Q&A]
Hi Jürgen,

I want to compare the regression coefficients of several single-case studies. However, the studies use different scales to measure the dependent variable. Therefore, the effect sizes are not comparable and standardization is necessary. [Van den Noortgate and Onghena (2008)](https://doi.org/10.1080/17489530802505362) suggest dividing the unstandardized regression coefficient by the standard error of the estimate (also known as the root mean squared error). For standardizing the sampling error variance of the regression coefficients, I would proceed analogously. I divide the sampling error variance by the estimated residual error variance. When I specify the regression model using the `lm()` function, standardization is possible. I tested this with a sample dataset `exampleAB$Johanna` from the *scan* package. Here is the R code for the standardization:

```{r}
lm <- lm(y ~ 1 + t + p + pt, dat)
summary(lm)
coef(lm)
coef(lm)/summary(lm)$sigma #standardizing regression coefficients
(coef(summary(lm))[, "Std. Error"])^2/summary(lm)$sigma^2 #standardizing sampling error variance
```

With the *scan* package, this was also possible, but a bit more cumbersome. However, I was not able to extract the standard errors of the regression coefficients. Consequently, I could not standardize the sampling error variance. Maybe you could help me with this. And is there a way to directly output the RMSE with the `plm()` function?
```{r message=FALSE, warning=FALSE}
library(scan)
plm <- plm(exampleAB$Johanna)
plm
plm[["full.model"]][["coefficients"]]/sqrt(1/16*plm[["full.model"]][["deviance"]]) #standardizing regression coefficients
```
I hope I have clearly explained my problem. Thank you very much for your help.

Max

**Comment by jazznbass** [comment]
Hi Max,

if you want standardized regression coefficients, the best way is to standardize the variables and then conduct the plm.
in the dev version of scan a new function (rescale()) is implemented that is quite convenient:
`exampleAB$Johanna |> rescale() |> plm()`
or with the released version:

```
exampleAB$Johanna |>   
  standardize(c("mt", "values"), scale = TRUE) |>   
  plm()
```

The standard errors are displayed in the output of the plm function and can be extrsacted with the coef() command:

```
> exampleAB$Johanna |> standardize(c("mt", "values"), scale = TRUE) |> plm() |> coef()
               Estimate Std. Error     t value    Pr(>|t|)
(Intercept) -1.27591010  0.3342017 -3.81778507 0.001514818
mt           0.05083088  0.8071737  0.06297391 0.950567295
phaseB       0.67518699  0.4997529  1.35104178 0.195479127
interB       0.77517095  0.8214611  0.94364905 0.359381347
```

Here, the regression weights are already standardized.

Best, Jürgen
